data-prepper-plugins/aws-lambda/src/main/java/org/opensearch/dataprepper/plugins/lambda/common/client/LambdaClientFactory.java

```diff
@@ -16,20 +16,21 @@
 public final class LambdaClientFactory {
     private LambdaClientFactory() { }
 
-    static LambdaClient createLambdaClient(final LambdaSinkConfig lambdaSinkConfig,
-                                           final AwsCredentialsSupplier awsCredentialsSupplier) {
-         final AwsCredentialsOptions awsCredentialsOptions = convertToCredentialsOptions(lambdaSinkConfig.getAwsAuthenticationOptions());
+    public static LambdaClient createLambdaClient(final AwsAuthenticationOptions awsAuthenticationOptions,
+                                                  final int maxConnectionRetries,
+                                                  final AwsCredentialsSupplier awsCredentialsSupplier) {
+         final AwsCredentialsOptions awsCredentialsOptions = convertToCredentialsOptions(awsAuthenticationOptions);
          final AwsCredentialsProvider awsCredentialsProvider = awsCredentialsSupplier.getProvider(awsCredentialsOptions);
 
             return LambdaClient.builder()
-                .region(lambdaSinkConfig.getAwsAuthenticationOptions().getAwsRegion())
+                .region(awsAuthenticationOptions.getAwsRegion())
```

Comment:
Consider enabling SDK metrics to track number of request, timeout, throttle etc.